rust-sdk/whirlpool/src/position.rs

```diff
@@ -295,3 +296,82 @@ pub async fn fetch_positions_in_whirlpool(
     let filters = vec![PositionFilter::Whirlpool(whirlpool)];
     fetch_all_position_with_filter(rpc, filters).await
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::tests::{
+        setup_ata_with_amount, setup_mint_with_decimals, setup_position, setup_whirlpool,
+        RpcContext,
+    };
+    use serial_test::serial;
+    use solana_program_test::tokio;
+    use std::error::Error;
+
+    #[tokio::test]
+    #[serial]
+    #[ignore = "Skipped until solana-bankrun supports gpa"]
+    async fn test_fetch_positions_for_owner_no_positions() -> Result<(), Box<dyn Error>> {
+        let ctx = RpcContext::new().await;
+        let owner = ctx.signer.pubkey();
+        let positions = fetch_positions_for_owner(&ctx.rpc, owner).await?;
+        assert!(
+            positions.is_empty(),
+            "No positions should exist for a new owner"
+        );
+        Ok(())
+    }
+
+    #[tokio::test]
+    #[serial]
+    #[ignore = "Skipped until solana-bankrun supports gpa"]
+    async fn test_fetch_positions_for_owner_with_position() -> Result<(), Box<dyn Error>> {
+        let ctx = RpcContext::new().await;
+        let mint_a = setup_mint_with_decimals(&ctx, 9).await?;
+        let mint_b = setup_mint_with_decimals(&ctx, 9).await?;
+        setup_ata_with_amount(&ctx, mint_a, 1_000_000_000).await?;
+        setup_ata_with_amount(&ctx, mint_b, 1_000_000_000).await?;
+
+        let whirlpool = setup_whirlpool(&ctx, mint_a, mint_b, 64).await?;
+        let position_pubkey = setup_position(whirlpool).await?;
```

Comment:
Might be nice to add a position bundle and te_position as well to check if `fetch_positions_for_owner` finds those positions as well.

Same goes from the `fetch_positions_in_pool` test